lib/active_merchant/billing/gateways/trans_first_transaction_express.rb

```diff
@@ -0,0 +1,597 @@
+
+          update_payment_method_request = build_xml_payment_update_request do |doc|
+            doc.tag! "v1:cust" do
+              doc.tag! "v1:contact" do
+                doc.tag! "v1:id", r.params["contact_id"]
+              end
+
+              doc.tag! "v1:pmt" do
+                doc.tag! "v1:id", token
+                doc.tag! "v1:type", 1 # update
+                add_payment_method(doc, payment_method)
+              end
+            end
+          end
+
+          r.process { commit(update_payment_method_request) }
+        end
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((<[^>]+pan>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<[^>]+sec>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<[^>]+id>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<[^>]+regKey>)[^<]+(<))i, '\1[FILTERED]\2')
+      end
+
+      private
+
+      CURRENCY_CODES = Hash.new{|h,k| raise ArgumentError.new("Unsupported currency: #{k}")}
+      CURRENCY_CODES["USD"] = "840"
+
+      def credit_card?(payment_method)
+        payment_method.respond_to?(:number)
+      end
+
+      def headers
+        {
+          "Content-Type" => "text/xml"
+        }
+      end
+
+      def commit(request)
+        raw_response = begin
+          ssl_post(url, request, headers)
+        rescue ActiveMerchant::ResponseError => e
+          e.response.body
+        end
+
+        response = parse(raw_response)
+
+        succeeded = success_from(response)
+
+        Response.new(
+          succeeded,
+          message_from(succeeded, response),
+          response,
+          error_code: error_code_from(succeeded, response),
+          authorization: authorization_from(response),
+          avs_result: AVSResult.new(code: response["AVSCode"]),
+          cvv_result: CVVResult.new(response["CVV2Response"]),
+          test: test?
+        )
+      end
+
+      def url
+        test? ? test_url : live_url
+      end
+
+      def parse(xml)
+        response = {}
+        doc = Nokogiri::XML(xml).remove_namespaces!
+
+        doc.css("Envelope Body *").each do |node|
+          # node.name is more readable, but uniq_name is occasionally necessary
+          uniq_name = [node.parent.name, node.name].join('_')
+          response[uniq_name] = node.text
+          response[node.name] = node.text
+        end
+
+        response
+      end
+
+      def success_from(response)
+        fault = response["Fault"]
+        approved_transaction = APPROVAL_CODES.include?(response["rspCode"])
+        found_contact = response["FndRecurrProfResponse"]
+
+        return !fault && (approved_transaction || found_contact)
+      end
+
+      def error_code_from(succeeded, response)
+        return if succeeded
+        response["errorCode"] || response["rspCode"]
+      end
+
+      def message_from(succeeded, response)
+        if succeeded
+          if response["rspCode"] == "10"
```

Comment:
We don't need to branch right here. Just have one "Succeeded" response message, and the caller can look at the `rspCode` field themselves if they to differentiate.